src/networkextension.cs

```diff
@@ -963,13 +1014,45 @@ interface NETunnelProviderManager
 		[Export ("loadAllFromPreferencesWithCompletionHandler:")]
 		[Async]
 		void LoadAllFromPreferences (Action<NSArray, NSError> completionHandler);
-	
-		[NullAllowed, Export ("copyAppRules")]
-		NEAppRule[] AppRules { get; }
-	
+
+		[NoWatch, NoTV, NoiOS, Mac (10,15,4)]
+		[Static]
+		[Export ("forPerAppVPN")]
+		NETunnelProviderManager CreatePerAppVpn ();
+
+		[return: NullAllowed]
+		[Export ("copyAppRules")]
+		NEAppRule[] CopyAppRules ();
```

Comment:
I wonder if needs a `[return: Release ()]`?

Reply:
You can’t do that because is a managed array there is no Handle

Reply:
We should fix the generator if it complains, because otherwise we'll leak.

Reply:
Yeah you are right, we are orphaning the native array, I'll look into it

Reply:
the generator code _should_ cope with that, if the attribute is present, and return the same managed array.

Reply:
It does not, I'll fix it

Reply:
I guess it could be done in another PR though, since it's not new API.

Reply:
Agreed